src/main/java/com/yelp/nrtsearch/server/luceneserver/search/TerminateAfterWrapper.java

```diff
@@ -77,7 +77,14 @@ public SearcherResult reduce(Collection<TerminateAfterCollectorWrapper> collecto
     if (didTerminateEarly) {
       onEarlyTerminate.run();
     }
-    return in.reduce(innerCollectors);
+    SearcherResult searcherResult = in.reduce(innerCollectors);
+    int totalHits = 0;
+    for (TerminateAfterCollectorWrapper collector : collectors) {
+      totalHits += collector.docCount;
+    }
+    searcherResult.getTopDocs().totalHits =
+        new TotalHits(totalHits, TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO);
+    return searcherResult;
```

Comment:
If there was not an early termination, you can skip doing the total hits rewrite.